### PR TITLE
Add additional index from schema functions

### PR DESCRIPTION
**Pull request by JHuculak**
As part of the Snapshot Migration work, we need to extract existing SchemaObject values from SchemaObjects representing the data on a component. While many of the Unreal-facing classes that represent SchemaObjects do have IndexFromSchema functions (which facilitates extracting collections from SchemaObjects) defined, the ones addressed in this PR did not.

The Migrator doesn't reason about cardinality of a field in Schema, since the Singular and Optional field cases can simply be treated as degenerate Arrays (with 1 and 0 or 1 elements, respectively). As such, it maintains a thin compatability wrapper that allows us to use IndexFromSchema functions to extract an arbitrary number of _correctly-typed_ objects from a SchemaObject. This type-correctness is important for two reasons: first, we may need to apply a transform (e.g., updating component Ids to accomodate schema changes) to the object before writing it back to the new snapshot); second, it also trivializes writing the object to the new SchemaObject, as AddToSchema functions already exist to handle the gruntwork of translating an Unreal-facing Component onto a SchemaObject.

#### Description
Adds `IndexFromSchema` functions for the following classes:
* `Coordinates`
* `WorkerRequirementSet`
* `FRotator`
* `FVector`
Also remaps the `GetFromSchema` functions for these classes to invoke the appropriate `IndexFromSchema` function internally.

#### Release note
Added `IndexYFromSchema` functions for the `Coordinates`, `WorkerRequirementSet`, `FRotator`, and `FVector` classes. Remapped the `GetYFromSchema` functions for the same classes to invoke `IndexYFromSchema` internally, in line with other implementations of the pattern.

#### Tests
Was able to use the new functions to successfully migrate a number of snapshots; existing CI tests continue to function.

#### Documentation
Release note

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Utils/SchemaUtils.h [hunk 1]
```diff
@@ -107,6 +107,10 @@ inline WorkerRequirementSet GetWorkerRequirementSetFromSchema(Schema_Object* Obj
 	return RequirementSet;
 }
 
+inline WorkerRequirementSet GetWorkerRequirementSetFromSchema(Schema_Object* Object, Schema_FieldId Id)
+{
+	return IndexWorkerRequirementSetFromSchema(Object, Id, 0);
+}
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/SchemaUtils.h]:
```suggestion
}

```

File: SpatialGDK/Source/SpatialGDK/Public/Schema/StandardLibrary.h [hunk 2]
```diff
@@ -55,9 +55,9 @@ inline void AddCoordinateToSchema(Schema_Object* Object, Schema_FieldId Id, cons
 	Schema_AddDouble(CoordsObject, 3, Coordinate.Z);
 }
 
-inline Coordinates GetCoordinateFromSchema(Schema_Object* Object, Schema_FieldId Id)
+inline Coordinates IndexCoordinateFromSchema(Schema_Object* Object, Schema_FieldId Id, uint32 Index)
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Public/Schema/StandardLibrary.h]: Looking at the c_worker.h
```
/* Functions that access a value by index for a particular field ID in an object. Boolean values are
 * guaranteed to be either 0 or 1.
 *
 * If the index doesn't exist for the given field, a default is returned; call
 * the corresponding GetCount function above to to determine if the total number of fields.
 *
 * Note that, for best performance, fields should be accessed in field ID and index order. */
```
I'm wondering about error handling here if an incorrect `Index` is used. I believe we should either do a `GetCount` before creating the `Schema_IndexObject` or check if the result is this `default`.

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDK/Public/Schema/StandardLibrary.h]: Then again, perhaps it's up to the caller of `IndexCoordinateFromSchema` to call `GetCount` before trying to index.... I'll go with that. Could we add a comment informing users that this is the recommended flow?

Comment (JHuculak) [reply SpatialGDK/Source/SpatialGDK/Public/Schema/StandardLibrary.h]: From what I can tell via uses of the various other `IndexYFromSchema` functions, I think there's an implicit expectation that:
a) The function is going to be invoked from within a loop that provides the `Index` argument and
b) The Index argument will exist in the range `[0, GetYCount...)`

Do we want to do a documentation pass over _all_ of the `IndexYFromSchema` functions noting that this is the recommended usage? Would it be worth doing that as a separate PR?

**Pull request merged**